Sources/PackagePlugin/PackageModel.swift

```diff
@@ -414,6 +414,30 @@ public enum FileType {
     case unknown
 }
 
+extension Package {
+    /// The list of targets matching the given names. Throws an error if any of
+    /// the targets cannot be found.
+    public func targets(named targetNames: [String]) throws -> [Target] {
+        return try targetNames.map { name in
+            guard let target = self.targets.first(where: { $0.name == name }) else {
+                throw PluginContextError.targetNotFound(name: name, package: self)
```

Comment:
consider returning none instead of throwing an error, tho probably a style thing

Reply:
Yeah, this seemed to be the common case.  There is possibly also room for a separate method that returns an optional, but it's less of a win for the common case,  and seems like something that can be added later.

Reply:
We can also improve the error here to include all the unknown names at once, but that wouldn't affect the API.  Totally open to change the API here though.

Reply:
I think its fine